platforms/macos/src/event.rs

```diff
@@ -3,17 +3,31 @@
 // the LICENSE-APACHE file) or the MIT license (found in
 // the LICENSE-MIT file), at your option.
 
-use accesskit::NodeId;
+use accesskit::{Live, NodeId};
 use accesskit_consumer::{DetachedNode, FilterResult, Node, TreeChangeHandler, TreeState};
-use objc2::foundation::NSString;
+use objc2::{
+    foundation::{NSInteger, NSMutableDictionary, NSNumber, NSString},
+    msg_send,
+    rc::Id,
+    Message,
+};
 use std::rc::Rc;
 
 use crate::{
     appkit::*,
     context::Context,
-    node::{filter, NodeWrapper},
+    node::{filter, filter_detached, NodeWrapper},
 };
 
+// Workaround for https://github.com/madsmtm/objc2/issues/306
+fn set_object_for_key<K: Message, V: Message>(
+    dictionary: &mut NSMutableDictionary<K, V>,
+    value: &V,
+    key: &K,
+) {
+    let _: () = unsafe { msg_send![dictionary, setObject:value forKey:key] };
```

Comment:
Nit: Add comma between arguments

```suggestion
    let _: () = unsafe { msg_send![dictionary, setObject: value, forKey: key] };
```